src/utils/request.ts

```diff
@@ -195,3 +196,68 @@ export function getRequestIP(
     return event.node.req.socket.remoteAddress;
   }
 }
+
+export interface RequestFingerprintOptions {
+  /** @default SHA-1 */
+  hash?: false | "SHA-1";
+
+  /** @default `true` */
+  ip?: boolean;
+
+  /** @default `false` */
+  xForwardedFor?: boolean;
+
+  /** @default `false` */
+  method?: boolean;
+
+  /** @default `false` */
+  path?: boolean;
+
+  /** @default `false` */
+  userAgent?: boolean;
+}
+
+/** @experimental Behavior of this utility might change in the future versions */
+export async function getRequestFingerprint(
+  event: H3Event,
+  opts: RequestFingerprintOptions = {},
+): Promise<string> {
+  const fingerprint: unknown[] = [];
+
+  if (opts.ip !== false) {
+    fingerprint.push(
+      getRequestIP(event, { xForwardedFor: opts.xForwardedFor }),
+    );
+  }
+
+  if (opts.method === true) {
+    fingerprint.push(event.method);
+  }
+
+  if (opts.path === true) {
+    fingerprint.push(event.path);
+  }
+
+  if (opts.userAgent === true) {
+    fingerprint.push(getRequestHeader(event, "user-agent"));
+  }
+
+  const fingerprintString =
+    fingerprint.filter(Boolean).join("|") ||
+    Math.random().toString(36).slice(2);
```

Comment:
```suggestion
  if (!fingerprint.length) {
    throw new RangeError('At least one fingerprinting factor must be enabled.')
  }

  const fingerprintString = fingerprint.filter(Boolean).join("|");
```

Reply:
I would rather have an error thrown here, warning the developer. Otherwise, they might end up with different fingerprint for the same request.
(Range error is somewhat applicable, but it can be an error too)

Reply:
We might instead disable random behavior and return null allowing usage to handle it for custom error wdty? (usually h3 utils are errorless)

Reply:
That's better too than a random string. In this case, the return value should be dependent on the options' argument type. Just so the user doesn't have to assert that the fingerprint is in fact a string.

Reply:
shall I make this change?